backend/types/request-User-Type.ts

```diff
@@ -0,0 +1,12 @@
+import { Request } from 'express';
+
+interface RequestUserType {
+  _id: string;
+  role: string;
+  userName: string;
+  email: string;
+}
+
+export interface RequestWithUserRole extends Request {
+  user?: RequestUserType;
```

Comment:
Do we actually need this? Can't we directly use the above ? make it both to type

Reply:
in middleware req.user , user is not found in Request type...extended the Request Type added user filed